microcloud/go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/canonical/microcloud/microcloud
 
-go 1.18
+go 1.21.4
```

Comment:
is this required for the change, why not go 1.22 like LXD?

Reply:
My laptop is running 1.21.4 (unlike my workstation that runs 1.22). I'll update this to 1.22 (alongside with my laptop go version)

Reply:
Was that (like in https://github.com/canonical/microcloud/actions/runs/8525721170/job/23353341519?pr=268):

```
 # github.com/canonical/microcluster/internal/db
Error: ../../../../go/pkg/mod/github.com/canonical/microcluster@v0.0.0-20240311124409-8d4c3238a91b/internal/db/db.go:134:21: not enough arguments in call to query.Retry
	have (func() error)
	want (context.Context, func(ctx context.Context) error)
make: *** [Makefile:9: build] Error 1
Error: Process completed with exit code 2.
```